openc3/spec/models/widget_model_spec.rb

```diff
@@ -0,0 +1,163 @@
+        expect(target["updated_at"]).to be_within(10_000_000).of(default_time)
+        target = WidgetModel.get(name: "OTHER", scope: 'scope')
+        expect(target["name"]).to eql "OTHER"
+        expect(target["updated_at"]).to be_within(10_000_000).of(other_time)
+      end
+    end
+
+    describe "handle_config" do
+      parser = OpenC3::ConfigParser.new("https://openc3.com")
+
+      it "only recognizes WIDGET" do
+        #parser = double("ConfigParser").as_null_object
+        expect(parser).to receive(:verify_num_parameters)
+        WidgetModel.handle_config(parser, "WIDGET", ["TEST_INT"], scope: "DEFAULT")
+        expect { WidgetModel.handle_config(parser, "DISABLE_ERB", ["TEST_INT"], scope: "DEFAULT") }.to raise_error(ConfigParser::Error)
+      end
+
+      it "self.handle_config" do
+        # blows up
+        # parms = ['name', 'label']
+        # model = WidgetModel.handle_config(parser, 'WIDGET', parms, scope: 'scope')
+        # expect(model).to be_a WidgetModel
+        expect{
+          WidgetModel.handle_config(parser, 'NOT_A_KEYWORD', ['some', 'parms'], scope: 'scope')
+        }.to raise_error(OpenC3::ConfigParser::Error)
+      end
+
+      it "handle_config" do
+        model = WidgetModel.new(name: "DEFAULT", scope: 'scope')
+        expect(model).to be_a WidgetModel
+        model.handle_config(parser, 'DISABLE_ERB', ['aParm'])
+        expect{
+          model.handle_config(parser, 'NOT_A_KEYWORD', ['other', 'parms'])
+        }.to raise_error(OpenC3::ConfigParser::Error)
+      end
+    end
+
+    describe "self.names" do
+      it "returns all widget names" do
+        model = WidgetModel.new(name: "TEST", scope: 'DEFAULT')
+        model.create
+        model = WidgetModel.new(name: "OTHER", scope: 'DEFAULT')
+        model.create
+        names = WidgetModel.names(scope: 'DEFAULT')
+        # contain_exactly doesn't care about ordering and neither do we
+        expect(names).to contain_exactly("TEST", "OTHER")
+      end
+    end
+
+    describe "self.all" do
+      it "returns all the parsed widgets" do
+        model = WidgetModel.new(name: "DEFAULT", scope: 'scope')
+        model.create
+        model = WidgetModel.new(name: "OTHER", scope: 'scope')
+        model.create
+        all = WidgetModel.all(scope: 'scope')
+        expect(all.keys).to contain_exactly("DEFAULT", "OTHER")
+      end
+    end
+
+    describe "as_json" do
+      it "encodes all the input parameters" do
+        model = WidgetModel.new(name: "DEFAULT", scope: 'scope', updated_at: 12345)
+        json = model.as_json(:allow_nan => true)
+        expect(json['name']).to eql "DEFAULT"
+        expect(json['updated_at']).to eql 12345
+      end
+    end
+
+    describe "deploy" do
+      it "unless validate_only" do
+        dir = File.join(SPEC_DIR, "install")
+        model = WidgetModel.new(name: "DEFAULT", scope: 'scope', updated_at: 12346)
+        model.create
+        model.deploy(dir, {}, validate_only: true)
+      end
```

Comment:
This isn't testing anything ... there are no expect statements

Reply:
Fixed